File: CompactGUI.Core/SharedObjects.vb
Public Class AnalysedFileDetails

    Public FileName As String
    Public UncompressedSize As Long
    Public CompressedSize As Long
    Public CompressionMode As CompressionAlgorithm
    Public FileInfo As IO.FileInfo
End Class


'Used to track efficiency of compression and built results for submission to wiki
Public Class ExtensionResult

    Public Property extension As String
    Public Property uncompressedBytes As Long
    Public Property compressedBytes As Long
    Public Property totalFiles As Integer
    ReadOnly Property cRatio As Decimal
        Get
            Return Math.Round(compressedBytes / uncompressedBytes, 2)
        End Get
    End Property

End Class

Public Enum CompressionAlgorithm
    NO_COMPRESSION = -2
    LZNT1 = -1
    XPRESS4K = 0
    LZX = 1
    XPRESS8K = 2
    XPRESS16K = 3
End Enum

File: CompactGUI.Core/SharedObjects.cs
namespace CompactGUI.Core;


public sealed class AnalysedFileDetails
{
    public required string FileName { get; set; }
    public long UncompressedSize { get; set; }
    public long CompressedSize { get; set; }
    public WOFCompressionAlgorithm CompressionMode { get; set; }
    public FileInfo? FileInfo { get; set; }
}


public sealed class ExtensionResult
{
    public required string Extension { get; set; }
    public long UncompressedBytes { get; set; }
    public long CompressedBytes { get; set; }
    public int TotalFiles { get; set; }
    public double CRatio => CompressedBytes == 0 ? 0 : Math.Round((double)CompressedBytes / UncompressedBytes, 2);

}




public struct CompressionProgress
{
    public int ProgressPercent;
    public string FileName;

    public CompressionProgress(int progressPercent, string fileName)
    {
        ProgressPercent = progressPercent;
        FileName = fileName;
    }
}


public enum CompressionMode: int
{
    XPRESS4K,
    XPRESS8K,
    XPRESS16K,
    LZX,
    None
}


public enum WOFCompressionAlgorithm: int
{
    NO_COMPRESSION = -2,
    LZNT1 = -1,
    XPRESS4K = 0,
    LZX = 1,
    XPRESS8K = 2,
    XPRESS16K = 3
}